nonce/nonce.go

```diff
@@ -247,17 +291,31 @@ func (ns *NonceService) Valid(nonce string) bool {
 	return true
 }
 
-func splitNonce(nonce string) (string, string, error) {
-	if len(nonce) < 4 {
+// splitDeprecatedNonce splits a nonce into a prefix and a body.
+//
+// Deprecated: Use NonceService.splitDeprecatedNonce instead.
+//
+// TODO(#6610): Remove this function once we've moved to derivable prefixes by
+// default.
+func splitDeprecatedNonce(nonce string) (string, string, error) {
+	if len(nonce) < DeprecatedPrefixLen {
+		return "", "", errInvalidNonceLength
+	}
+	return nonce[:DeprecatedPrefixLen], nonce[DeprecatedPrefixLen:], nil
+}
+
+// splitNonce splits a nonce into a prefix and a body.
+func (ns *NonceService) splitNonce(nonce string) (string, string, error) {
```

Comment:
It is slightly confusing to have a NonceService method sandwiched in between two non-method functions (splitDeprecatedNonce and RemoteRedeem).